src/libmongoc/tests/TestSuite.c

```diff
@@ -831,19 +827,16 @@ TestSuite_PrintJsonHeader(TestSuite *suite, /* IN */
            "    \"apiVersion\": %s\n"
            "  },\n"
            "  \"results\": [\n",
-           egetenv("MONGOC_TEST_USER"),
-           egetenv("MONGOC_TEST_PASSWORD"),
-           hostname,
-           port,
-           egetenv("MONGOC_TEST_URI"),
-           egetenv("MONGOC_TEST_GSSAPI_HOST"),
-           egetenv("MONGOC_TEST_GSSAPI_USER"),
-           udspath,
+           egetenv("MONGOC_TEST_USER") ? "(set)" : "(unset)",
+           egetenv("MONGOC_TEST_PASSWORD") ? "(set)" : "(unset)",
+           egetenv("MONGOC_TEST_URI") ? "(set)" : "(unset)",
```

Comment:
Do we need to find other potential instances of logging the URI and related credentials? e.g. `uri_str` contains both `test_frame_get_user_password()` and `test_framework_get_host_and_port()`: https://github.com/mongodb/mongo-c-driver/blob/2198bf238d5486b2c5793509d88e64775a9bbf8a/src/libmongoc/tests/test-libmongoc.c#L886-L889